libs/model/src/featurelayer.cpp

```diff
@@ -83,11 +83,138 @@ TileFeatureLayer::TileFeatureLayer(
 
 TileFeatureLayer::~TileFeatureLayer() = default;
 
+namespace
+{
+
+/* Check that starting from a given index, the parts of an id composition
+ * match the featureIdParts segment from start for the given length.
+ */
+bool idPartsMatchComposition(
+    std::vector<IdPart> const& candidateComposition,
+    uint32_t compositionMatchStartIdx,
+    KeyValuePairs const& featureIdParts,
+    unsigned long matchLength)
+{
+    auto featureIdIter = featureIdParts.begin();
+    auto compositionIter = candidateComposition.begin();
+
+    while (compositionMatchStartIdx > 0) {
```

Comment:
FYI, there is a neat syntax trick you can do here: `while (compositionMatchStartIndex --> 0)`

Not recommending it though, because it makes people think that there is a `-->` operator.

Reply:
Yes, it's good to know... if it's commonly used, we could change it in the entire codebase in the future for style consistency.